src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker_IVsSolutionLoadEvents.cs

```diff
@@ -51,6 +95,217 @@ int IVsSolutionLoadEvents.OnAfterBackgroundSolutionLoadComplete()
+                var referencedProject = ImmutableProjects.SingleOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.ProjectFilePath, projectReferencePath));
+                if (referencedProject == null)
+                {
+                    referencedProject = (AbstractProject)CreateProjectFromArgumentsAndReferences(
+                        workspaceProjectContextFactory,
+                        projectReferencePath,
+                        allProjectInfos,
+                        targetPathsToProjectPaths);
+                }
+
+                var referencedProjectContext = referencedProject as IWorkspaceProjectContext;
+                if (referencedProjectContext != null)
+                {
+                    addedProjectReferences.Add(projectReferencePath);
+                    projectContext.AddProjectReference(
+                        referencedProjectContext,
+                        new MetadataReferenceProperties());
+                }
+                else if (referencedProject != null)
+                {
+                    // This project was already created by the regular project system. See if we
+                    // can find the matching project somehow.
+                    var existingReferenceOutputPath = referencedProject?.BinOutputPath;
+                    if (existingReferenceOutputPath != null)
+                    {
+                        addedProjectReferences.Add(projectReferencePath);
+                        projectContext.AddMetadataReference(
+                            existingReferenceOutputPath,
+                            new MetadataReferenceProperties());
+                    }
+                }
+                else
+                {
+                    // We don't know how to create this project.  Another language or something?
+                }
+            }
+
+            foreach (var reference in commandLineArguments.MetadataReferences)
+            {
+                string possibleProjectReference;
+                if (targetPathsToProjectPaths.TryGetValue(reference.Reference, out possibleProjectReference) &&
+                    addedProjectReferences.Contains(possibleProjectReference))
+                {
+                    // We already added a P2P reference for this, we don't need to add the file reference too.
+                    continue;
+                }
+
+                projectContext.AddMetadataReference(reference.Reference, reference.Properties);
+            }
+
+            foreach (var reference in commandLineArguments.AnalyzerReferences)
+            {
+                projectContext.AddAnalyzerReference(reference.FilePath);
+            }
+
+            return projectContext;
+        }
+
+        private static string GetLanguageOfProject(string projectFilename)
+        {
+            switch (Path.GetExtension(projectFilename))
+            {
+                case ".csproj":
```

Comment:
DPL force loads projects like venus, and xproj doesn't exist in Dev15.  I believe this should be safe.